spec/models/topic_spec.rb

```diff
@@ -1362,6 +1362,61 @@ def set_state!(group, user, state)
         expect(user.user_profile.dismissed_banner_key).to be_nil
       end
 
+      context "until date" do
+        context "valid date" do
+          it "enqueues a future job to remove it" do
+            bannered_until = 5.days.from_now
+
+            expect(topic.archetype).to eq(Archetype.default)
+
+            expect do
+              topic.make_banner!(user, bannered_until.to_s)
+            end.to change { Jobs::RemoveBanner.jobs.size }.by(1)
+
+            expect(topic.archetype).to eq(Archetype.banner)
+
+            job = Jobs::RemoveBanner.jobs[0]
+            expect(Time.at(job["at"])).to be_within_one_minute_of(bannered_until)
+            expect(job["args"][0]["topic_id"]).to eq(topic.id)
+
+            job["class"].constantize.new.perform(*job["args"])
+            topic.reload
+
+            expect(topic.archetype).to eq(Archetype.default)
```

Comment:
I don't think we should be testing the `Jobs::RemoveBanner` job in this spec here. I would create a spec file specifically to test the job for better organisation.

Reply:
Agree, we should always test jobs with an independent spec